File: C++/130B-Honest Coach.cpp
/*
    Solution to Codeforces problem 130B
    Copyright (c) GeneralMing. All rights reserved.

    https://github.com/GeneralMing/codeforces
*/

#include<iostream>
#include<string>
#include <bits/stdc++.h>

using namespace std;

int main()
{
	int t, n;
	cin >> t;
	while(t--)
	{
		int min = INT_MAX;
		cin >> n;
		int athletes[n];
		for(int i = 0; i < n; i++)
			cin >> athletes[i];
		sort(athletes, athletes + n);
		for(int i = 0; i < n-1; i++)
			if(min > abs(athletes[i] - athletes[i+1]))
				min = abs(athletes[i] - athletes[i+1]);
		cout << min << endl;
	}
}

File: Python/130B-Honest Coach.py
"""
    Solution to Codeforces problem 130B
    Copyright (c) GeneralMing. All rights reserved.

    https://github.com/GeneralMing/codeforces
"""

t = int(input())
while(t != 0):
	min = 1001
	n = int(input())
	athletes = input().split(" ")
	for i in range(n):
		athletes[i] = int(athletes[i])
	athletes.sort()
	for i in range(n-1):
		if(min > abs(athletes[i] - athletes[i+1])):
			min = abs(athletes[i] - athletes[i+1])
	print(min)
	t -= 1